In [0]:
from pyspark.sql import functions as F
from pyspark.sql import Row
import pandas as pd

In [0]:
filename = "s3://full-stack-bigdata-datasets/Big_Data/Project_Steam/steam_game_output.json"

In [0]:
df = spark.read.format("json").load(filename)
df.printSchema()

root
 |-- data: struct (nullable = true)
 |    |-- appid: long (nullable = true)
 |    |-- categories: array (nullable = true)
 |    |    |-- element: string (containsNull = true)
 |    |-- ccu: long (nullable = true)
 |    |-- developer: string (nullable = true)
 |    |-- discount: string (nullable = true)
 |    |-- genre: string (nullable = true)
 |    |-- header_image: string (nullable = true)
 |    |-- initialprice: string (nullable = true)
 |    |-- languages: string (nullable = true)
 |    |-- name: string (nullable = true)
 |    |-- negative: long (nullable = true)
 |    |-- owners: string (nullable = true)
 |    |-- platforms: struct (nullable = true)
 |    |    |-- linux: boolean (nullable = true)
 |    |    |-- mac: boolean (nullable = true)
 |    |    |-- windows: boolean (nullable = true)
 |    |-- positive: long (nullable = true)
 |    |-- price: string (nullable = true)
 |    |-- publisher: string (nullable = true)
 |    |-- release_date: string (nullable = true)
 |    |-

In [0]:
df.count()

Out[5]: 55691

## Which publisher hast released the most games on Steam ?

In [0]:

# get data
df = df.withColumn("publisher", F.col("data").getField("publisher"))

# processing
publisher_counts_df = df.groupBy("publisher").count().orderBy(F.col("count").desc())

# visualisation
display(publisher_counts_df)


publisher count Big Fish Games 422 8floor 202 SEGA 165 Strategy First 151 Square Enix 141 Choice of Games 140 Sekai Project 132 HH-Games 132 132 Ubisoft 127 Laush Studio 126 THQ Nordic 125 Alawar Entertainment 107 Fulqrum Publishing 104 Plug In Digital 101 Ziggurat 100 Slitherine Ltd. 99 Sokpop Collective 99 Devolver Digital 98 KOEI TECMO GAMES CO., LTD. 90 Reforged Group 89 Electronic Arts 88 Komodo 87 PLAYISM 87 SA Industry 87 Microids 84 Piece Of Voxel 81 Hosted Games 79 Kagura Games 79 Boogygames Studios 78 Artifex Mundi 75 tinyBuild 72 Daedalic Entertainment 72 Blender Games 71 BANDAI NAMCO Entertainment 69 Paradox Interactive 69 Tero Lunkka 68 2K 67 My Way Games 66 ESDigital Games 64 Garage Games 62 MAGIX Software GmbH 62 Nacon 60 Forever Entertainment S. A. 60 RewindApp 59 Ripknot Systems 55 United Independent Entertainment 55 Bethesda Softworks 55 Back To Basics Gaming 54 Hede 54 MangaGamer 53 GrabTheGames 52 William at Oxford 52 Focus Entertainment 52 Activision 51 KEMCO 51 Ghost_RUS Games 50 Aldorlea Games 50 Individual Software 49 Sometimes You 49 Trinity Project 47 Volens Nolens Games 47 玫瑰工作室 47 Iceberg Interactive 46 Droid Riot 46 Atomic Fabrik, Cristian Manolachi 46 PsychoFlux Entertainment 46 Xbox Game Studios 46 Funbox Media Ltd 45 ImperiumGame 44 NIS America, Inc. 44 PQube 44 Dnovel 43 Kalypso Media Digital 43 HexWar Games 42 Enoops 42 8Floor 41 Pixel Games UK 41 505 Games 41 Atari 40 Big Black Bear 40 HandyGames 40 Winged Cloud 40 familyplay 40 Anamik Majumdar 40 Deep Silver 38 HandMade Games 37 Team17 Digital Ltd 37 Rebellion 37 New Reality Games 37 Disney 37 kovalevviktor 37 Headup 37 Bitlock Studio 37 Zoo Corporation 36 MumboJumbo 36 Piko Interactive LLC, Bleem! 35 GameHouse 35 Milestone S.r.l. 35 Valve 35 Qumaron 35 Capcom 35 khukhrovr 35 Luci Entertainment 35 EnsenaSoft 35 Curve Games 35 Aerosoft GmbH 34 Jetdogs Studios 34 DigiPen Institute of Technology 34 KISS ltd 33 Spike Chunsoft Co., Ltd. 33 Meridian4 33 Kedexa 32 For Kids 32 Humongous Entertainment, Nightdive Studios 32 ANPA.US 32 HeR Interactive 32 Assemble Entertainment 31 Gotcha Gotcha Games 31 NanningsGames 31 Arc System Works 31 Outright Games Ltd. 31 Nightdive Studios 31 Fruitbat Factory 30 LucasArts, Lucasfilm, Disney 29 Minimol Games 29 SakuraGame 29 Warner Bros. Interactive Entertainment 29 Raw Fury 29 Louie Inc 29 橙光游戏 29 Bohemia Interactive 28 INFINITE BRIDGE 28 kazakovstudios 28 Snkl Studio 28 Good Shepherd Entertainment 28 Armor Games Studios 28 OTAKU Plan 28 Viva Media 27 EpiXR Games UG 27 Team17 27 Gamera Games 27 Winter Wolves 26 Kairosoft Co.,Ltd 26 Anatoliy Loginovskikh 26 Humble Games 26 ebi-hime 26 XSEED Games, Marvelous USA, Inc. 26 КиКо 25 KPL 25 Quarlellle 25 Grey Alien Games 25 Konami Digital Entertainment 25 Digerati 25 SNK CORPORATION 25 Henteko Doujin 25 Ratalaika Games S.L. 25 Digital Tribe 25 Freedom Games 25 Wolverine Studios 24 Jackbox Games, Inc. 24 Cute Hannah's Games 24 CFK Co., Ltd. 24 Paradise Project 24 CAPCOM Co., Ltd. 24 Versus Evil 24 Idea Factory International 24 FIVE-BN GAMES 24 Retroism, Nightdive Studios 24 Interplay Entertainment Corp. 24 Quiet River 24 Libredia 24 astragon Entertainment 24 Frontwing USA 24 MIRROR Soft 23 Annapurna Interactive 23 indie_games_studio 23 Gamenesis 23 W.T.B. 23 None 22 SOFTSTAR ENTERTAINMENT 22 Sacada 22 Warfare Studios 22 LTZinc 22 Senpai Studios 22 Phoenixxx Games 22 My Label Game Studio 22 Cosmi Valusoft 21 Alternative Software Ltd 21 RunServer 21 Cristian Manolachi, Atomic Fabrik 21 XSEED Games, Marvelous USA, Inc., Marvelous 21 No Gravity Games 21 MACKINN7 21 Dear Villagers 20 Headup Games 20 Codemasters, Electronic Arts 20 Stardock Entertainment 20 Ultimate Games S.A. 20 AFBIK Studio 20 Metal Fox 20 HIKARI FIELD 20 Telltale Games 20 Meng Games 20 Archor Games 20 Mens Sana Interactive 20 Tonguç Bodur 20 Red twice potato 20 Lovely Games 20 Rex Junior 20 DLsite 20 Casual Arts 20 PopCap Games, Inc. 19 Nicalis, Inc. 19 Enaayah Software Development and Services Private Limi

Big Fish Games is the publisher who published the most games on Steam (422).

## What are the best rated games?

## What are the most represented languages?

In [0]:
# get data
languages_df = df.withColumn("languages", F.col("data").getField("languages"))

# pre-processing
languages_df = languages_df.withColumn("split_languages", F.split(F.col("languages"),",\s*"))
languages_df = languages_df.withColumn("language", F.explode(F.col("split_languages")))

# processing
languages_df = languages_df.groupBy("language").count().orderBy(col("count").desc())

# visualisation
display(languages_df)

language count English 55116 German 14019 French 13426 Russian 12922 Simplified Chinese 12782 Spanish - Spain 12233 Japanese 10368 Italian 9304 Portuguese - Brazil 6750 Korean 6600 Traditional Chinese 6307 Polish 5376 Portuguese - Portugal 4011 Turkish 3602 Dutch 3080 Spanish - Latin America 2732 Czech 2352 Swedish 2047 Ukrainian 1928 Hungarian 1923 Arabic 1796 Danish 1717 Thai 1699 Finnish 1614 Norwegian 1613 Greek 1444 Romanian 1431 Not supported 1276 Bulgarian 1256 Vietnamese 1133 Indonesian 54 Slovak 47 Catalan 38 Hindi 26 Hebrew 25 Belarusian 23 Serbian 20 Basque 17 Malay 14 Persian 12 Croatian 12 Lithuanian 12 Latvian 12 Portuguese 11 11 Estonian 10 Filipino 7 Slovenian 6 Welsh 6 Galician 6 Icelandic 5 Irish 5 Georgian 4 English[b]*[/b] 4 Afrikaans 4 Albanian 4 Bangla 3 Urdu 3 Bosnian 3 Tamil 3 Kazakh 3 German[b]*[/b] 2 Telugu 2 Luxembourgish 2 #lang_slovakian 2 German; 2 Swahili 2 Azerbaijani 2 Kannada 1 Russian
[b]*[/b] 1 French[b]*[/b] 1 Japanese 

[b]*[/b] 1 English (full audio) 1 English Dutch English 1 Mongolian 1 Maori 1 Spanish - Spain
[b]*[/b] 1 Italian 

[b]*[/b] 1 Valencian 1 Dari 1 Marathi 1 Korean[b]*[/b] 1 Spanish - Spain[b]*[/b] 1 English
Russian
Spanish - Spain
Japanese
Czech 1 Traditional Chinese (text only) 1 Macedonian 1 Punjabi (Gurmukhi) 1 Uzbek 1 Italian[b]*[/b] 1 Russian
English
Spanish - Spain
French
Japanese
Czech 1 (all with full audio support) 1

## Are most games available on Windows/Mac/Linux?

In [0]:
# get data
platform_df = df.withColumn("platforms", F.col("data").getField("platforms"))\
    .withColumn("Linux", F.col("platforms.Linux"))\
    .withColumn("Mac",F.col("platforms.Mac"))\
    .withColumn("Windows",F.col("platforms.Windows"))\
    .withColumn("genre",F.col("data").getField("genre"))\
    .drop("id")  

# processing
count_mac = platform_df.filter(F.col('Mac') == "True").count()
count_linux = platform_df.filter(F.col('Linux') == "True").count()
count_windows = platform_df.filter(F.col('Windows') == "True").count()
total_count = df.count()

mac_percentage = int((count_mac / total_count) * 100)
linux_percentage = int((count_linux / total_count) * 100)
windows_percentage = int((count_windows / total_count) * 100)

# visualisation
print(f"There are {mac_percentage}% games on MacOS, {linux_percentage}% games on Linux, and {windows_percentage}% games available on Windows.")


There are 22% games on MacOS, 15% games on Linux, and 99% games available on Windows.
